src/components/DatasetSearchListItem/README.md

```diff
@@ -0,0 +1,132 @@
+# DatasetSearchListItem Component
+
+A TypeScript React component for displaying dataset search results with support for themes, dates, and navigation.
+
+## Props
+
+```typescript
+type SearchItemProps = {
+  title: string;                    // Dataset title
+  modified: string;                 // Modified date
+  description: string;              // Dataset description
+  identifier: string;               // Dataset identifier
+  downloadUrl?: string | null;      // Download URL (optional)
+  largeFile: boolean;               // Whether file is large
+  paginationEnabled: boolean;       // Whether pagination is enabled
+  dataDictionaryLinks: boolean;     // Whether to show data dictionary links
+  refresh?: string;                 // Refresh date (optional)
+  released?: string;                // Released date (optional)
+  showTopics?: boolean;             // Whether to show themes (default: false)
+  showDateDetails?: boolean;        // Whether to show detailed dates (default: false)
+  theme?: string[];                 // Array of theme strings (optional)
+  location: LocationType;           // Current location object (required)
+}
+```
+
+## Location Type
+
+```typescript
+type LocationType = {
+  pathname: string;                 // Current pathname (required)
+  search?: string;                  // Search query (optional)
+  hash?: string;                    // Hash fragment (optional)
+  state?: any;                      // Location state (optional)
+};
+```
+
+## Usage Examples
+
+### Basic Usage
+
+```tsx
+import { DatasetSearchListItem } from '@civicactions/cmsds-open-data-components';
+
+const location = {
```

Comment:
We should use useLocation from react-router-dom for this - as we are doing elsewhere in this library. That means we probably don't have to pass it around as a prop and can just use it in-component as needed

Reply:
addressed
https://github.com/GetDKAN/cmsds-open-data-components/pull/293/commits/5493a5db7d68e610b337870fc70c79a7fe26d2c6